src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,369 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+
+import { Extendable } from '../../../types/utils'
+import { ComponentSlotStyle } from '../../themes/types'
+import Downshift, { DownshiftState, StateChangeOptions } from 'downshift'
+import Label from '../Label/Label'
+import { UIComponent, customPropTypes } from '../../lib'
+import Input from '../Input/Input'
+import keyboardKey from 'keyboard-key'
+import List from '../List/List'
+import Text from '../Text/Text'
+import Image from '../Image/Image'
+import ListItem from '../List/ListItem'
+import _ from 'lodash'
+import Icon from '../Icon/Icon'
+
+export interface DropdownProps {
+  className?: string
+  fluid?: boolean
+  items?: DropdownListItem[]
+  multiple?: boolean
+  onChange?: (active: DropdownListItem[]) => any
+  placeholder?: string
+  search?: boolean
+  styles?: ComponentSlotStyle<DropdownProps, DropdownState>
+  toggleButton?: boolean
+}
+
+export interface DropdownState {
+  active: DropdownListItem[]
+  backspaceDelete: boolean
+  focused: boolean
+  inputValue?: string
+  message?: string
+}
+
+export interface DropdownListItem {
+  key: string
+  header: string
+  content?: string
+  image?: string
+}
+
+/**
+ */
+export default class Dropdown extends UIComponent<Extendable<DropdownProps>, DropdownState> {
+  private inputRef: HTMLElement
+
+  static displayName = 'Dropdown'
+
+  static className = 'ui-dropdown'
+
+  static propTypes = {
+    /** Additional CSS class name(s) to apply.  */
+    className: PropTypes.string,
+
+    /** An input can take the width of its container. */
+    fluid: PropTypes.bool,
+
+    /** Shorthand array of props for ListItem. */
+    items: customPropTypes.collectionShorthand,
+
+    /** A dropdown can have a multiple selection. */
+    multiple: PropTypes.bool,
+
+    /**
+     * Event for request to change 'open' value.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props and proposed value.
+     */
+    onChange: PropTypes.func,
+
+    /** A message to serve as placeholder. */
+    placeholder: PropTypes.string,
+
+    /** A dropdown can have a search field. */
+    search: PropTypes.bool,
+
+    /** Additional CSS styles to apply to the component instance.  */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** A dropdown can have a toggle button. */
+    toggleButton: PropTypes.bool,
```

Comment:
`toggleButton` does not work for me. Clicking on it when dropdown is open closes it but then it is immediately reopened.

Reply:
it has an issue right now, when it is open, it will close and then open it again. I am still thinking on a proper fix.

Downshift checks if the button is part of the `combobox` div and if not, it will open the list. But I had to remove both the list and the toggle button from the `combobox` div because of screen reader navigation issue. And that's how the bug appeared.

Still thinking on how to fix this.